paymentsheet/src/main/java/com/stripe/android/paymentsheet/injection/StripeCustomerAdapterComponent.kt

```diff
@@ -0,0 +1,114 @@
+import com.stripe.android.paymentsheet.repositories.CustomerEphemeralKeyProvider
+import com.stripe.android.paymentsheet.repositories.CustomerRepository
+import com.stripe.android.paymentsheet.repositories.SetupIntentClientSecretProvider
+import com.stripe.android.paymentsheet.repositories.StripeCustomerAdapter
+import dagger.Binds
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import java.util.Calendar
+import javax.inject.Named
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+
+internal const val CAN_CREATE_SETUP_INTENT = "canCreateSetupIntent"
+
+@Singleton
+@Component(
+    modules = [
+        StripeCustomerAdapterModule::class,
+        StripeRepositoryModule::class,
+        CoroutineContextModule::class,
+        CoreCommonModule::class,
+    ]
+)
+@OptIn(ExperimentalSavedPaymentMethodsApi::class)
+internal interface StripeCustomerAdapterComponent {
+    val stripeCustomerAdapter: StripeCustomerAdapter
+
+    @Component.Builder
+    interface Builder {
+        @BindsInstance
+        fun context(context: Context): Builder
+
+        @BindsInstance
+        fun customerEphemeralKeyProvider(
+            customerEphemeralKeyProvider: CustomerEphemeralKeyProvider
+        ): Builder
+
+        @BindsInstance
+        fun setupIntentClientSecretProvider(
+            setupIntentClientSecretProvider: SetupIntentClientSecretProvider?
+        ): Builder
+
+        @BindsInstance
+        fun canCreateSetupIntents(
+            @Named(CAN_CREATE_SETUP_INTENT) canCreateSetupIntents: Boolean
+        ): Builder
+
+        fun build(): StripeCustomerAdapterComponent
+    }
+}
+
+@Module
+@OptIn(ExperimentalSavedPaymentMethodsApi::class)
+internal interface StripeCustomerAdapterModule {
+    @Binds
+    fun bindsCustomerRepository(repository: CustomerApiRepository): CustomerRepository
+
+    companion object {
+        @Provides
+        fun provideTimeProvider(): () -> Long = {
+            Calendar.getInstance().timeInMillis
+        }
+
+        @Provides
+        fun providePrefsRepositoryFactory(
+            appContext: Context,
+            @IOContext workContext: CoroutineContext
+        ): (CustomerEphemeralKey) -> PrefsRepository = { customer ->
+            DefaultPrefsRepository(
+                appContext,
+                customer.customerId,
+                workContext
+            )
+        }
+
+        @Provides
+        fun providePaymentConfiguration(appContext: Context): PaymentConfiguration {
+            return PaymentConfiguration.getInstance(appContext)
+        }
+
+        @Provides
+        @Named(PUBLISHABLE_KEY)
+        fun providePublishableKey(configuration: PaymentConfiguration): () -> String = {
+            configuration.publishableKey
+        }
+
+        @Provides
+        @Named(PRODUCT_USAGE)
+        fun providesProductUsage(): Set<String> = setOf("WalletMode")
```

Comment:
@wooj-stripe is this the same as iOS?

Reply:
I"m guessing that StripeCustomerAdapterModule is similar to our protocol/interface called CustomerAdapter.  If that is the case, we don't have a "ProductUsage" field -- can this field just be removed?

Reply:
@wooj-stripe  I was thinking if we wanted to track product usage (similar to how we do in PaymentSheet here: https://github.com/stripe/stripe-ios/blob/e6f7515faf404ee99a5c0d5b790c85f629391694/StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet.swift#L91)

Reply:
Thanks for flagging this -- I'll add this on my list of things to investigate, and will update our code to match what android has assuming it makes sense.

Reply:
It's an easy change on our end too! Just making sure we're at least somewhat aligned.